contracts/JBSingleTokenPaymentTerminalStore3_2.sol

```diff
@@ -582,25 +585,23 @@ contract JBSingleTokenPaymentTerminalStore3_2 is
     // Get a reference to the new used overflow allowance for this funding cycle configuration.
     uint256 _newUsedOverflowAllowanceOf = usedOverflowAllowanceOf[
       IJBSingleTokenPaymentTerminal(msg.sender)
-    ][_projectId][fundingCycle.configuration] + _amount;
+    ][_projectId][fundingCycle.configuration][_currency] + _amount;
 
     // There must be sufficient allowance available.
-    (uint256 _overflowAllowanceOf, uint256 _overflowAllowanceCurrency) = IJBController3_2(
-      directory.controllerOf(_projectId)
-    ).fundAccessConstraintsStore().overflowAllowanceOf(
+    uint256 _overflowAllowance = IJBController3_2(directory.controllerOf(_projectId))
+      .fundAccessConstraintsStore()
+      .overflowAllowanceOf(
         _projectId,
         fundingCycle.configuration,
         IJBSingleTokenPaymentTerminal(msg.sender),
-        IJBSingleTokenPaymentTerminal(msg.sender).token()
+        IJBSingleTokenPaymentTerminal(msg.sender).token(),
+        _currency
       );
 
     // Make sure the new used amount is within the allowance.
-    if (_newUsedOverflowAllowanceOf > _overflowAllowanceOf || _overflowAllowanceOf == 0)
+    if (_newUsedOverflowAllowanceOf > _overflowAllowance || _overflowAllowance == 0)
       revert INADEQUATE_CONTROLLER_ALLOWANCE();
 
-    // Make sure the currencies match.
-    if (_currency != _overflowAllowanceCurrency) revert CURRENCY_MISMATCH();
```

Comment:
Note for reviewers: no need to check this any longer since the distributionLimit for _currency is explicitly sought after.